GitUI/BuildServerIntegration/BuildServerWatcher.cs

```diff
@@ -82,6 +81,14 @@ public async Task LaunchBuildServerInfoFetchOperationAsync()
             var fromNowObservable = buildServerAdapter.GetFinishedBuildsSince(scheduler, DateTime.Now);
 
             var isBuildRunning = false;
+
+            // We are only interested by the 'OnComplete' which occurs:
+            // * after 10s if a build is running
+            // * after (NumberOfPollSkipWhenNoBuildsAreRunning * 10s) if no builds are running
+            var delayObservable = Observable.Interval(TimeSpan.FromSeconds(10))
+                .TakeWhile(a => !isBuildRunning && a < NumberOfPollSkipWhenNoBuildsAreRunning)
+                .SkipWhile(_ => true).Select(_ => new BuildInfo());
```

Comment:
Yes, SkipWhile() is needed because we are just interested by the duration of the observable before it terminates and not the items emitted (that's why they are skipped).

The 'Select()' is just there to have an observable of the good type but will never be used.

The real value is in the 'TakeWhile()' that:

* terminates the observable early (after 10s) if there was a build running found, to call the api again and update the build status.
* Or let the observable terminates later (after receiving 12 items in the observable, each one separated by 10s) to poll the build status less often when no running builds are found.

Reply:
> The 'Select()' is just there to have an observable of the good type but will never be used.

So why do we need to allocate then?

Reply:
>So why do we need to allocate then?

We won't. Because there will never be an item due to the 'SkipWhile(true)'. The new is just there to infer the type.

Or I really missed something and there is another way?

Reply:
tested, repeats after short interval if any builds observed, otherwise repeats after long interval

```csharp
            bool anyBuilds = false;
            var cancellationToken = new CompositeDisposable
                    {
                        fullDayObservable.OnErrorResumeNext(fullObservable)
                                         .OnErrorResumeNext(Observable.Empty<BuildInfo>()
                                                                      .DelaySubscription(TimeSpan.FromMinutes(1))
                                                                      .OnErrorResumeNext(fromNowObservable)
                                                                      .Retry()
                                                                      .Repeat())
                                         .ObserveOn(MainThreadScheduler.Instance)
                                         .Subscribe(OnBuildInfoUpdate),

                        runningBuildsObservable
                            .Do(buildInfo => anyBuilds = true)
                            .Retry()
                            .Concat(
                                Observable.Defer(() => Observable.Empty<BuildInfo>().DelaySubscription(
                                    anyBuilds
                                        ? ShortPollInterval
                                        : LongPollInterval)))
                            .Finally(() => anyBuilds = false)
                            .Repeat()
                            .ObserveOn(MainThreadScheduler.Instance)
                            .Subscribe(OnBuildInfoUpdate2)
                    };
```